packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -144,13 +150,20 @@ export class TerminalFrontendContribution implements CommandContribution, MenuCo
         }
     }
 
-    protected async newTerminal(): Promise<void> {
-        const widget = <TerminalWidget>await this.widgetManager.getOrCreateWidget(TERMINAL_WIDGET_FACTORY_ID, <TerminalWidgetFactoryOptions>{
-            created: new Date().toString()
+    async newTerminal(options: TerminalWidgetOptions): Promise<TerminalWidget> {
+        const widget = <TerminalWidgetImpl>await this.widgetManager.getOrCreateWidget(TERMINAL_WIDGET_FACTORY_ID, <TerminalWidgetFactoryOptions>{
+            created: new Date().toString(),
+            ...options
         });
-        this.shell.addWidget(widget, { area: 'bottom' });
-        this.shell.activateWidget(widget.id);
-        widget.start();
+        return widget;
     }
 
+    activateWidget(widget: TerminalWidget): void {
+        const tabBar = this.shell.getTabBarFor(widget);
+        if (!tabBar) {
+            this.shell.expandPanel('bottom');
+            this.shell.addWidget(widget, { area: 'bottom' });
+            this.shell.activateWidget(widget.id);
```

Comment:
Should not we activate always regardless whether the widget was already attached or not?